tests/ceph_installer/test_cephadm.py

```diff
@@ -49,6 +57,75 @@
 )
 
 
+def get_host_osd_map(cls):
+    out, err = cls.shell(args=["ceph", "osd", "tree"])
+    osd_string = out
+    osd_dict = {}
+    for osd_str in osd_string.split("host")[1:]:
+        osd = [i.strip() for i in osd_str.splitlines()]
+        osd_dict[osd[0]] = list()
+        for i in osd[1:]:
+            temp = i.split()
+            for x in temp:
+                if "osd" in x:
+                    osd_dict[osd[0]].append(x)
+    return osd_dict
+
+
+def validate_log_file_after_enable(cls):
+    """
+    Method to verify generation of log files in default log directory when logging not enabled
+    Args:
+        cls: cephadm instance object
+
+    Returns:
+        string
+    """
+    out, err = cls.shell(
+        args=["ceph", "config", "set", "global", "log_to_file", "true"]
+    )
+    out, err = cls.shell(args=["ceph", "fsid"])
+    fsid = out.strip()
+    log_file_path = os.path.join("/var/log/ceph", fsid)
+    roles_to_validate = ["mon", "mgr", "osd"]
+    osd_dict = get_host_osd_map(cls)
+
+    for node in cls.cluster.get_nodes():
+        try:
+            if "client" in node.role and len(node.role) == 1:
+                continue
+            out, _ = node.exec_command(
+                cmd=f"ls -l {log_file_path} | egrep -v '^d'", sudo=True
+            )
+            files = out.read().decode().split("\n")
+            files = list(filter(None, files))
+            LOG.info("Files in log folder : %s" % ", ".join(files))
+            roles = []
+            [
+                roles.append(role)
+                for role in node.role
+                if role not in roles and role in roles_to_validate
+            ]
+            for role in roles:
+                if role == "osd":
```

Comment:
Maybe we could avoid if we go with daemon names and construct the log file path.

Reply:
Like mentioned in the above comment, it is not possible to construct the complete log file path in case of VMs.

Reply:
Made the necessary changes.